crates/bitwarden/src/tool/generators/username_forwarders/addyio.rs

```diff
@@ -0,0 +1,63 @@
+use reqwest::{header::CONTENT_TYPE, StatusCode};
+
+use crate::error::{Error, Result};
+pub async fn generate(
+    http: &reqwest::Client,
+    api_token: String,
+    domain: String,
+    base_url: String,
+    website: Option<String>,
+) -> Result<String> {
+    if api_token.is_empty() {
+        return Err(Error::Internal("Invalid addy.io API token."));
+    }
+    if domain.is_empty() {
+        return Err(Error::Internal("Invalid addy.io domain."));
+    }
+    if base_url.is_empty() {
+        return Err(Error::Internal("Invalid addy.io url."));
+    }
```

Comment:
This is a bit ugly. Would we benefit from making a NonEmptyString with a try/from into a string?

Reply:
I've added these as they also were there on the current clients code. I'm wondering if they make any sense at all here, though. Using `"!&%$"` as domain is just as invalid as `""`, maybe we can just remove the checks entirely and let the API request fail. What do you think?

Reply:
Sounds good to me.